### DISCUSSION TITLE
base branch issues - custom ADO script - in onboarding PR phase

### DISCUSSION THREAD

**Discussion by patrickd77-eng** [renovatebot/renovate#26041] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest as of 29/11/23

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### What is your question?

I have a specific requirement, not sure if it can work out.

My repo structure is as follows 
main -> develop -> various branches. I'm currently in my own branch, based on develop. In this branch I have created a YAML pipeline, renovate.json and .npmrc. I've been following [this guide](https://docs.renovatebot.com/modules/platform/azure/) roughly.

When running the pipeline, I can see it would work OK and the suggestions look good. It's in the phase where it creates an onboarding branch and PR. I know how to reset this process if required (renaming PR and deleting branch).

My requirement is that I absolutely cannot have it merging into 'main' - we will do that on an ad hoc basis. I want it to create its branch, suggestions and merge into 'develop'. Even for its first attempt, it **must** PR into 'develop'. To be clear I don't need automerge, we'll still review.

I have tried various solutions but really struggling to find something that actually stops it using main as the baseBranch.  To be clear, there is no config or yaml at all outside of my local branch, at the moment. I'm just manually running the pipeline in ADO. Would be grateful for your suggestions. Do I need to have a renovate.json file already up in main or something?

Here is my YAML, config etc.

YAML
`
schedules:
  - cron: "0 0 1 * *"  # At 00:00 on day-of-month 1
    displayName: "Monthly Dependency Update on the 1st Day"
    branches:
      include: [develop]
    always: true
variables:
  - group: REDACTED
trigger: none  # Disables CI trigger
pool:
  vmImage: ubuntu-latest
steps:
  - script: npm install -g renovate
    displayName: 'Install Renovate Globally'
  - task: npmAuthenticate@0
    inputs:
      workingFile: 'Deployments/RenovateBot/.npmrc'
  - bash: |
      git config --global user.email 'bot@renovateapp.com'
      git config --global user.name 'Renovate Bot'
      npx --userconfig Deployments/RenovateBot/.npmrc renovate
    displayName: 'Run Renovate Bot'
    env:
      RENOVATE_PLATFORM: azure
      RENOVATE_ENDPOINT: $(System.CollectionUri)
      RENOVATE_TOKEN: $(REDACTED)
      RENOVATE_REPOSITORIES: 'REDACTED'
      RENOVATE_BASE_BRANCH: 'develop' 
`

.npmrc
`registry=REDACTED 
always-auth=true`

renovate.json
`{
  "baseBranch": "develop"
}`

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You must have a config in your default branch, but your `baseBranches` config can exclude the default branch. In other words, you don't have to have any PRs targeting the default branch, but you need to have a config

**Reply by patrickd77-eng** [reply]
Thanks, so the renovate.json needs to be checked in to main with the basebranch setting I listed above, and then the expected behaviour should happen?

**Reply by rarkins** [reply]
Yes, I think all you need is:

```
{
  "baseBranches": ["develop"],
  "useBaseBranchConfig": true
}
```

The second setting means that you can then put your "real" config in the develop branch if you want to. Even if you don't want to do that (e.g. because you merge develop into main regularly), it still allows you to _modify_ your behavior in developer without committing to main.

**Reply by jovinson-ms** [reply]
As of 4/18/2025, `true` is no longer a supported value for `useBaseBranchConfig`: https://docs.renovatebot.com/configuration-options/#usebasebranchconfig
